src/backend/mpi/mpi.cpp

```diff
@@ -220,25 +222,24 @@ namespace argo {
 		namespace atomic {
 			void _exchange(global_ptr<void> obj, void* desired,
 					std::size_t size, void* output_buffer) {
-				sem_wait(&ibsem);
 				MPI_Datatype t_type = fitting_mpi_int(size);
 				// Perform the exchange operation
-				MPI_Win_lock(MPI_LOCK_EXCLUSIVE, obj.node(), 0, globalDataWindow[0]);
-				MPI_Fetch_and_op(desired, output_buffer, t_type, obj.node(), obj.offset(), MPI_REPLACE, globalDataWindow[0]);
-				MPI_Win_unlock(obj.node(), globalDataWindow[0]);
-				// Cleanup
-				sem_post(&ibsem);
+				std::size_t win_index = get_data_win_index(obj.offset());
+				std::size_t win_offset = get_data_win_offset(obj.offset());
+				// TODO: Does this have to be to node 0 like in prev implementation?
```

Comment:
This seems to me more of an XXX-style comment rather than a TODO.
In any case, perhaps *this* is the time to resolve this question.